libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/ChargingStationsPreserver.kt

```diff
@@ -0,0 +1,64 @@
+package com.mapbox.navigation.core.reroute
+
+import com.google.gson.JsonObject
+import com.google.gson.JsonPrimitive
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.core.routeoptions.isEVRoute
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal suspend fun preserveChargingStationsFromOldRoutes(
```

Comment:
Now I think it's better to move this workaround to router wrapper until NN supports it on their side

Reply:
It's better to leave it here, because I will need one more client side response update: NAVAND-1429. Leaving all logic here will let me create only one extra copy of the route.

Reply:
I should also mention a disadvantage: initial offline route response won't have charging station metadata, but it will be fixed here: https://mapbox.atlassian.net/browse/NN-854

Reply:
Why? It's only applicable for reroutes, right? I think it's fine in reroute controller.

Reply:
> It's only applicable for reroutes, right? 

It's also nice to have for offline route request with user provided waypoints, isn't it?

Reply:
Oh, right. But we won't need it once NN support it on their side, right?